azure-iot-device/azure/iot/device/provisioning/provisioning_device_client.py

```diff
@@ -0,0 +1,40 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module provides an abstract interface representing clients which can communicate with the
+Device Provisioning Service.
+"""
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ProvisioningDeviceClient(object):
+    """
+    Super class for any client that can be used to register devices to Device Provisioning Service.
+    """
+
+    def __init__(self, transport):
+        """
+        Initializes the registration client.
+        :param transport: Instance of the Transport object.
+        """
+        self._transport = transport
+
+    @abc.abstractmethod
+    def register(self):
+        """
+        Register the device with the Device Provisioning Service.
```

Comment:
>Register the device with the Device Provisioning Service. [](start = 8, length = 57)

This comment should be updated to include more information on special cases that we just talked about, such as when does this function return, what happens if the registration fails, what happens if the user calls this a second time while the registration is in progress, or a second time after the registration is complete, or a second time after the registration failed.  It should also specify _some_ of the details on what it does beyond registering, such as "This connects to the registration service, sends a registration request, waits for a successful registration or an error, and then disconnects."